client/state/plugins/installed/test/selectors-ts.ts

```diff
@@ -0,0 +1,193 @@
+					},
+				},
+				[ siteTwoId ]: {
+					'akismet/akismet': {
+						status: 'error',
+						action: INSTALL_PLUGIN,
+						error: createError( 'no_package', 'Download failed.' ),
+					},
+				},
+			},
+		},
+	},
+	sites: {
+		items: {
+			[ siteOneId ]: {
+				ID: siteOneId,
+				jetpack: true,
+				visible: true,
+				name: 'One Site',
+			},
+			[ siteTwoId ]: {
+				ID: siteTwoId,
+				jetpack: true,
+				visible: true,
+				name: 'Two Site',
+			},
+		},
+	},
+	...userState,
+} );
+
+const jetpackWithSites = {
+	id: 'jetpack/jetpack',
+	slug: 'jetpack',
+	name: 'Jetpack by WordPress.com',
+	plugin_url: 'http://jetpack.com',
+	description:
+		'Bring the power of the WordPress.com cloud to your self-hosted WordPress. Jetpack enables you to connect your blog to a WordPress.com account to use the powerful features normally only available to WordPress.com users.',
+	author: 'Automattic',
+	author_url: 'http://jetpack.com',
+	network: false,
+	sites: {
+		[ siteTwoId ]: {
+			active: true,
+			autoupdate: true,
+			version: '4.1.1',
+			update: {
+				compatibility: {
+					scalar: {
+						scalar: false,
+					},
+				},
+				id: '20101',
+				new_version: '4.2.2',
+				package: 'https://downloads.wordpress.org/plugin/jetpack.4.2.2.zip',
+				plugin: 'jetpack/jetpack.php',
+				slug: 'jetpack',
+				tested: '4.6',
+				url: 'https://wordpress.org/plugins/jetpack/',
+			},
+		},
+	},
+};
+
+const akismetWithSites = {
+	id: 'akismet/akismet',
+	slug: 'akismet',
+	name: 'Akismet',
+	plugin_url: 'https://akismet.com/',
+	description:
+		'Used by millions, Akismet is quite possibly the best way in the world to <strong>protect your blog from spam</strong>. It keeps your site protected even while you sleep. To get started: 1) Click the "Activate" link to the left of this description, 2) <a href="https://akismet.com/get/">Sign up for an Akismet plan</a> to get an API key, and 3) Go to your Akismet configuration page, and save your API key.',
+	author: 'Automattic',
+	author_url: 'https://automattic.com/wordpress-plugins/',
+	network: false,
+	sites: {
+		[ siteOneId ]: {
+			active: true,
+			autoupdate: true,
+			version: '3.1.11',
+		},
+	},
+};
+
+const helloDollyWithSites = {
+	id: 'hello-dolly/hello',
+	slug: 'hello-dolly',
+	name: 'Hello Dolly',
+	plugin_url: 'https://wordpress.org/plugins/hello-dolly/',
+	description:
+		'This is not just a plugin, it symbolizes the hope and enthusiasm of an entire generation summed up in two words sung most famously by Louis Armstrong: Hello, Dolly. When activated you will randomly see a lyric from <cite>Hello, Dolly</cite> in the upper right of your admin screen on every page',
+	author: 'Matt Mullenweg',
+	author_url: 'http://ma.tt/',
+	network: false,
+	sites: {
+		[ siteOneId ]: {
+			active: false,
+			version: '1.6',
+			autoupdate: true,
+		},
+		[ siteTwoId ]: {
+			active: false,
+			version: '1.6',
+			autoupdate: true,
+		},
+	},
+};
+
+describe( 'getAllPluginsIndexedByPluginSlug', () => {
```

Comment:
The following scenarios appear to be already tested in the unit tests:
* we're requesting plugins of a site
* with the same plugin in more than 1 site.

However they're not represented by their own separate tests. Is this something that makes sense to have separately tested, since it's a substantial part of the selector functionality?

Furthermore, should we also cover the case when we're requesting plugins for all sites?

Finally, should we introduce a few tests that verify that the cache is working fine?

Reply:
That was a mistake on my part, I accidentally put the assertions in a single test. I split them out now, and I added tests for the memoization.

The case for requesting all sites isn't covered by these selectors, so it will be in the next PR.